jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/nodiagnostics/NoDiagnosticsTest.java

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+* Copyright (c) 2023 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4jakarta.jdt.nodiagnostics;
+
+import static org.eclipse.lsp4jakarta.jdt.core.JakartaForJavaAssert.assertJavaDiagnostics;
+
+import java.util.Arrays;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.lsp4jakarta.commons.JakartaDiagnosticsParams;
+import org.eclipse.lsp4jakarta.jdt.core.BaseJakartaTest;
+import org.eclipse.lsp4jakarta.jdt.core.JDTUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class NoDiagnosticsTest extends BaseJakartaTest {
+
+	@Parameter
+	public String filePath;
+
+	protected static JDTUtils JDT_UTILS = new JDTUtils();
+
+	@Test
+	public void checkForNoDiagnostics() throws Exception {
+
+		JDTUtils utils = JDT_UTILS;
+
+		IJavaProject javaProject = loadJavaProject("jakarta-noDiagnostic-sample", "");
+		IFile javaFile = javaProject.getProject()
+				.getFile(new Path(filePath));
+		String uri = javaFile.getLocation().toFile().toURI().toString();
+
+		JakartaDiagnosticsParams diagnosticsParams = new JakartaDiagnosticsParams();
+		diagnosticsParams.setUris(Arrays.asList(uri));
+
+		// should be no diagnostics in the file.
+		assertJavaDiagnostics(diagnosticsParams, utils);
+
+	}
+
+	// list of java files in demo-servlet-no-diagnostics.
+	@Parameters
+	public static String[] projectFileProvider() throws Exception {
+
+		String basePath = "src/main/java/io/openliberty/sample/jakarta/";
+		String files[] = { basePath + "annotations/GeneratedAnnotation.java",
```

Comment:
I think we should see if there is a way to dynamically retrieve the Java files from the "no-diagnostics" project. It will avoid having to modify this file every time new diagnostic test project files are added.